DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -57,31 +96,47 @@ function Get-TargetResource
             $($LocalizedData.FirewallRuleDoesNotExistMessage) -f $Name
             ) -join '')
         return @{
-            Name   = $Name
             Ensure = 'Absent'
+            Name   = $Name
         }
     }
 
     $properties = Get-FirewallRuleProperty -FirewallRule $firewallRule
 
-    # Populate the properties for get target resource
-    return @{
-        Name            = $Name
-        Ensure          = 'Present'
-        DisplayName     = $firewallRule.DisplayName
-        Group           = $firewallRule.Group
-        DisplayGroup    = $firewallRule.DisplayGroup
-        Enabled         = $firewallRule.Enabled
-        Action          = $firewallRule.Action
-        Profile         = $firewallRule.Profile.ToString() -replace(' ', '') -split(',')
-        Direction       = $firewallRule.Direction
-        Description     = $firewallRule.Description
-        RemotePort      = @($properties.PortFilters.RemotePort)
-        LocalPort       = @($properties.PortFilters.LocalPort)
-        Protocol        = $properties.PortFilters.Protocol
-        Program         = $properties.ApplicationFilters.Program
-        Service         = $properties.ServiceFilters.Service
+    $Result = @{
+        Ensure = 'Present'
+    }
+
+    # Populate the properties for get target resource by looping through
+    # the parameter array list and adding the values to 
+    foreach ($p in $ParameterList)
+    {
+
+        if ($p.type -eq 'Array')
```

Comment:
There is one tiny tiny difference between the two - but it is an important one (and super easy to over look):
```powershell
$Value = @(Invoke-Expression -Command "`$($($parameter.source))")
```
vs.
```powershell
$Value = (Invoke-Expression -Command "`$($($parameter.source))")
```
For the array (the first one) I have to explicitly cast this to an Array because otherwise if there is only a single array element PS will cast it to the type that the item is rather than keep it as an array.

So originally I just had it the if/then logic and no logging at all but I was trying to track down an issue occurring with the Get-DSCConfiguration logic so added that verbose logging. It really isn't needed anymore. So if you want I'm happy to get rid of the logs (but leave the if/then).